src/newgidmap.c

```diff
@@ -196,6 +196,12 @@ int main(int argc, char **argv)
 		return EXIT_FAILURE;
 	}
 
+	/* Validate fds coming from get_pidfd_from_fd() */
+	if (st.st_dev != 22 || st.st_rdev != 0) {
```

Comment:
Hm, on my laptop it's 23.  That bodes ill.

Reply:
There's another way to get these values. I'll update the patchset. One sec.

Reply:
I pushed the new version. Please, review.

Reply:
Looks great, thanks.

I'll take another look in 2m and then merge.

I can't tell you how embarrassed I am.